### PR TITLE
issue: 1946579 Fix event notification logic used by extraAPI

### PR DESCRIPTION
**Pull request by rafiw**
Signed-off-by: Rafi Wiener <rafiw@mellanox.com>

### CODE REVIEW

File: src/vma/dev/cq_mgr.cpp [hunk 1]
```diff
@@ -853,6 +853,34 @@ int cq_mgr::drain_and_proccess(uintptr_t* p_recycle_buffers_last_wr_id /*=NULL*/
...
+		if (res == 0) {
+			++cq_ev_count;
+		}
+	} while (res == 0);
+	if (errno != EAGAIN) {
+		return -1;
+	}
+	if (cq_ev_count > 0) {
+		ibv_ack_cq_events(m_p_ibv_cq, cq_ev_count);
+		return 1;
```
Comment (rafiw) [inline src/vma/dev/cq_mgr.cpp]: @ilansmith

File: src/vma/dev/ring_simple.h [hunk 2]
```diff
@@ -104,7 +104,7 @@ class ring_simple : public ring_slave
 	struct ibv_comp_channel* get_tx_comp_event_channel() { return m_p_tx_comp_event_channel; }
 	int			get_ring_descriptors(vma_mlx_hw_device_data &data);
 	void			modify_cq_moderation(uint32_t period, uint32_t count);
-
+	int			ack_and_arm_cq(cq_type_t cq_type);
```
Comment (igor-ivanov) [inline src/vma/dev/ring_simple.h]: Please consider removing `request_notification_blocking()` after introducing `ack_and_arm_cq()`. request_notification_blocking() was added just for vma_modify_ring() at https://github.com/Mellanox/libvma/commit/6b19d6e21ae4736b961242146dd4c1e151be3ec7.

File: src/vma/dev/cq_mgr.cpp [hunk 3]
```diff
@@ -853,6 +853,34 @@ int cq_mgr::drain_and_proccess(uintptr_t* p_recycle_buffers_last_wr_id /*=NULL*/
...
+		res = ibv_get_cq_event(m_comp_event_channel, &ib_cq, &cq_context);
+		if (res == 0) {
+			++cq_ev_count;
+		}
+	} while (res == 0);
+	if (errno != EAGAIN) {
+		return -1;
+	}
+	if (cq_ev_count > 0) {
+		ibv_ack_cq_events(m_p_ibv_cq, cq_ev_count);
```
Comment (igor-ivanov) [inline src/vma/dev/cq_mgr.cpp]: Probably calling `get_cq_event()` `cq_ev_count` times is needed as were done
See https://github.com/Mellanox/libvma/blob/master/src/vma/dev/cq_mgr.cpp#L945-L951
or
https://github.com/Mellanox/libvma/blob/master/src/vma/dev/cq_mgr.cpp#L904-L917

RM: https://github.com/Mellanox/libvma/commit/9c3b43e05797cfc5e66d03dfb41fb6b5b6c30298#diff-2170492bb09a82a83f97f7f158592e7c

Comment (rafiw) [reply src/vma/dev/cq_mgr.cpp]: void vma_ib_mlx5_get_cq_event(vma_ib_mlx5_cq_t *)
{
	// no need in operation with cq_sn as far as it is managed by driver code for now
}

Comment (rafiw) [reply src/vma/dev/cq_mgr.cpp]: it doe's nothing.
is there a test for the event?

File: src/vma/dev/ring_simple.h (hunk 2 above)
Comment (rafiw) [reply src/vma/dev/ring_simple.h]: good catch
done

File: src/vma/dev/cq_mgr.cpp (hunk 3 above)
Comment (rafiw) [reply src/vma/dev/cq_mgr.cpp]: done

Comment (igor-ivanov) [reply src/vma/dev/cq_mgr.cpp]: get_cq_event() just increment to 1. In your case you should call get_cq_event() cq_ev_count times.

Comment (rafiw) [reply src/vma/dev/cq_mgr.cpp]: done

**Pull request merged**